### PR TITLE
Update testing to Cloud Build

### LINKED ISSUE #262: Update testing to Cloud Build
We should update the testing to use Cloud Build and a new Service Account per run, as documented [here](https://docs.google.com/document/d/1eiHpZlhEDMDG8gi4LHyv2aZs1TJmeFlF8sqCE57_j4A/edit#heading=h.rz8l5qn0ygc0).

### PR DESCRIPTION
**Pull request by paulpalamarchuk**
* Fixed #262
 * Removed softlinks to tfvars from tests
 * Set `gsuit` provider version same for all examples
 * Disable test that require G-Suite (issue # 275)

### CODE REVIEW

**Comment by paulpalamarchuk** [comment]
Integration test results (minimal):

**Comment by thefirstofthe300** [comment]
As someone who has previously contributed to the CFT and now looks on from outside the CFT dev team, would it be possible to get the Cloud Build logs either exported to Github or allow view access to these builds for external users?

This would help demystify why builds are failing for external users who wish to contribute.

**Comment by morgante** [comment]
@thefirstofthe300 Unfortunately there's not currently any viable mechanism for externalizing Cloud Build logs.

However, I am happy to add anyone interested as a viewer if you provide your GCP email.

**Comment by paulpalamarchuk** [comment]
@morgante tests updated.

File: Makefile [hunk 1]
```diff
@@ -18,180 +18,68 @@
...
-		-e GROUP_NAME \
-		-e ADMIN_ACCOUNT_EMAIL \
-		-e ORG_ID \
-		-e PROJECT_ID \
 		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && exec /bin/bash"
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` here? i.e. `-v "$(CURDIR)":/workspace`. This is necessary to run the tests on my machine.

File: Makefile [hunk 2]
```diff
@@ -18,180 +18,68 @@
...
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen create"
-
-.PHONY: docker_converge
-docker_converge: ## Run `kitchen converge` within the Docker test environment
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` here? i.e. `-v "$(CURDIR)":/workspace`. This is necessary to run the tests on my machine.

File: Makefile [hunk 3]
```diff
@@ -18,180 +18,68 @@
...
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen converge"
-
-.PHONY: docker_verify
-docker_verify: ## Run `kitchen verify` within the Docker test environment
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` here? i.e. `-v "$(CURDIR)":/workspace`. This is necessary to run the tests on my machine.

File: Makefile [hunk 4]
```diff
@@ -18,180 +18,68 @@
...
-		-e GROUP_NAME \
-		-e ADMIN_ACCOUNT_EMAIL \
-		-e ORG_ID \
-		-e PROJECT_ID \
 		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen verify"
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` here? i.e. `-v "$(CURDIR)":/workspace`. This is necessary to run the tests on my machine.

File: Makefile [hunk 5]
```diff
@@ -18,180 +18,68 @@
...
-		-e GROUP_NAME \
-		-e ADMIN_ACCOUNT_EMAIL \
-		-e ORG_ID \
-		-e PROJECT_ID \
-		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen destroy"
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` here? i.e. `-v "$(CURDIR)":/workspace`. This is necessary to run the tests on my machine.

File: Makefile [hunk 6]
```diff
@@ -18,180 +18,68 @@
...
-		-e GROUP_NAME \
-		-e ADMIN_ACCOUNT_EMAIL \
-		-e ORG_ID \
-		-e PROJECT_ID \
-		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		make test_integration
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` here? i.e. `-v "$(CURDIR)":/workspace`. This is necessary to run the tests on my machine.

File: Makefile [hunk 7]
```diff
@@ -18,180 +18,68 @@
...
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen converge"
-
-.PHONY: docker_verify
-docker_verify: ## Run `kitchen verify` within the Docker test environment
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
```
Comment (Jberlinsky) [inline Makefile]: This should be expanded to include:

- `TF_VAR_gsuite_admin_email`
- `TF_VAR_gsuite_domain`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+```
+export SERVICE_ACCOUNT_JSON=$(< credentials.json)
+```
+
+You will also need to set a few environment variables:
+```
+export TF_VAR_org_id="your_org_id"
+export TF_VAR_folder_id="your_folder_id"
+export TF_VAR_billing_account="your_billing_account_id"
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply CONTRIBUTING.md]: Variables added to export

File: Makefile [hunk 9]
```diff
@@ -18,180 +18,68 @@
...
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen create"
-
-.PHONY: docker_converge
-docker_converge: ## Run `kitchen converge` within the Docker test environment
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply Makefile]: Variables added to export

File: Makefile (hunk 1 above)
Comment (paulpalamarchuk) [reply Makefile]: Updated

File: Makefile (hunk 2 above)
Comment (paulpalamarchuk) [reply Makefile]: Updated

File: Makefile (hunk 3 above)
Comment (paulpalamarchuk) [reply Makefile]: Updated

File: Makefile (hunk 4 above)
Comment (paulpalamarchuk) [reply Makefile]: Updated

File: Makefile (hunk 5 above)
Comment (paulpalamarchuk) [reply Makefile]: Updated

File: Makefile (hunk 6 above)
Comment (paulpalamarchuk) [reply Makefile]: Updated

**Comment by Jberlinsky** [comment]
@paulpalamarchuk Could you please rebase this to eliminate conflicts before we proceed with a review?

**Comment by paulpalamarchuk** [comment]
@Jberlinsky PR rebased.

File: Makefile (hunk 1 above)
Comment (aaron-lane) [reply Makefile]: @Jberlinsky can you open an issue for this against the module template?

**Comment by aaron-lane** [comment]
@paulpalamarchuk I resolved conflicts in the CHANGELOG. In general, we should only document changes which affect the public behaviour of the module; changes to testing do not need to be included.

File: Makefile (hunk 1 above)
Comment (Jberlinsky) [reply Makefile]: Yep! https://github.com/terraform-google-modules/terraform-google-module-template/issues/33

File: Makefile (hunk 9 above)
[the remark this answers is not in the document]
Comment (kopachevsky) [reply Makefile]: where?

[the remark this answers is not in the document]
Comment (aaron-lane) [reply Makefile]: @kopachevsky Below 👇

File: test/scripts/preconditions/test_preconditions.py [hunk 10]
```diff
@@ -1,153 +0,0 @@
-#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply test/scripts/preconditions/test_preconditions.py]: Reverted

File: test/helpers/test_migrate.py [hunk 11]
```diff
@@ -1,290 +0,0 @@
-#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply test/helpers/test_migrate.py]: Reverted

**Pull request merged**